### ISSUE TITLE
Zendesk API change - help center update article endpoint no longer allows changes to body, title or draft

### ISSUE THREAD

**Issue by Marcus10110** [mozts2005/ZendeskApi_v2#215]
Some time in the last 3 weeks, calls to HelpCenter.Articles.UpdateArticle can no longer update the title, body or draft parameters of a article. (it was known to work normally up to and including March 30th, and we discovered the issue on April 19th, 2016)

This can easily be shown - in our testing, changes to "label_names" would be properly saved, but changes to "title", "body", and "draft" are not saved.
We tested "title" and "body" but not the "draft" variable. [This is assumed from the Zendesk API documentation here](https://developer.zendesk.com/rest_api/docs/help_center/articles#update-article).

Note the line that says:

> The endpoints do not update translation properties such as the article's title, body or draft. See Translations.

We tested the translation endpoint with curl, and we were able to update the title there. We did not explicitly test the body or draft variables, but I assume they will work as normal.

This API wrapper does not currently support the translation methods, and support for them should be added to allow the modification of article body, title and draft variables.

I have not tested Sections or Categories yet. Here are relevent notes from the documentation:

[Zendesk API - Help Center - Sections - Update Section](https://developer.zendesk.com/rest_api/docs/help_center/sections#update-section)

> These endpoints only update section-level metadata such as the sorting position. They don't update section translations. See Translations.

[Zendesk API - Help Center - Category - Update Category](https://developer.zendesk.com/rest_api/docs/help_center/categories#update-category)

> These endpoints only update category-level metadata such as the sorting position. They don't update category translations.

**Comment by Marcus10110** [comment]
Ok, so I forked the repository and added some basic functionality. However, it's pretty rough and needs more work it should be merged in.

https://github.com/Marcus10110/ZendeskApi_v2

New commit:

https://github.com/Marcus10110/ZendeskApi_v2/commit/610acba78d86af6f165e60432fc6964ee045b42b

First, I added the ability to sideload translations with the normal get article call:

ZendeskApi.HelpCenter.Articles.GetArticle(long articleId, ArticleSideLoadOptionsEnum sideloadOptions = ArticleSideLoadOptionsEnum.None );

Previously, translations could only be loaded via using one of the GetArticles* functions, which returns more than one analyzer at a time.

Second, I've set up a Translations class on the help center, with these abilities:

- List missing translations for and individual article, section, or category.
- List all enabled locales for the help center, and also retrieve the default locale.
- Update a specific translation for an article, section or category.
- Delete a specific translation.

Still missing are these API calls:

- list translations for a specific article, section or category.
- show a specific translation for an article, section or category.


In addition to those calls missing from the Translations class, the Categories and Sections classes need to add the ability to sideload translations when getting a specific category or section by Id.

Also, I'd like to apologize - my visual studio has different code formatting defaults set up, so spacing everywhere won't match the rest of the project. :( If any code is ever used from my repo, feel free to reformat the whole thing.

I also forgot to say this last time - thanks so much for developing and maintaining such an awesome project! This nuget package has been invaluable.

**Comment by Marcus10110** [comment]
So, it turns there are two existing unit tests that fail due to this issue:

- CanCreateUpdateAndDeleteArticles
- CanCreateUpdateAndDeleteArticlesAsync

There are a few other tests that failed, but they seem unrelated.

I would like to get this fixed properly, but I'm a bit conflicted on the solution.

At the minimum, I've already added most of the required functions to update translations. However, it's not intuitive to the developer that UpdateArticle will not update the body, title, or draft flag on the article.

I can see three options:
1. we do nothing, except maybe mention this caveat in the comments.
2. In the update Article function, we do extra logic and optionally call update translation. (However, this would add a lot of overhead, and not be in the spirit of a light API-wrapper)
3. We could create a helper function to handle option #2, something to automatically update the default locale when updating an article.
3. yell at Zendesk to change this. (and still add translation support though)

I've had a ticket open with Zendesk about this for a while, and I just followed up with them, asking them to forward my feedback to the API team.
